File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.js
function solution(arr, n) {
    
    let result = [...arr];
    
    if (arr.length % 2 === 1) {
        for (let i = 0; i < arr.length; i += 2) {
            result[i] += n
        }
    } else {
        for (let i = 1; i < arr.length; i+= 2) {
            result[i] += n
        }
    }
    
    return result;
}

File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.py
def solution(arr, n):
    answer = []
    if len(arr) % 2 == 1:  # arr 길이가 홀수라면
        for i in range(len(arr)):
            if i % 2 == 0:     # 짝수 인덱스
                answer.append(arr[i] + n)
            else:
                answer.append(arr[i])
    else:  # arr 길이가 짝수라면
        for i in range(len(arr)):
            if i % 2 == 1:     # 홀수 인덱스
                answer.append(arr[i] + n)
            else:
                answer.append(arr[i])
    return answer
